### ISSUE TITLE
Schema Dump missing "IF NOT EXISTS" in create schema statement

### ISSUE THREAD

**Issue by SampsonCrowley** [rails/rails#38695]
### Steps to reproduce

1) Add `public` as one of the searchable schemas to `schema_search_path` in `database.yml`
  - e.g. `schema_search_path: auditing,other,public`
2) set `config.active_record.schema_format = :sql` in `application.rb

### Expected behavior

Because default schemas like `public` already exist, the `CREATE SCHEMA` statement should only attempt to create schemas that don't already exist:

```SQL
CREATE SCHEMA IF NOT EXISTS auditing;
CREATE SCHEMA IF NOT EXISTS other;
CREATE SCHEMA IF NOT EXISTS public;
```

### Actual behavior

Only a `CREATE SCHEMA` statement is dumped. This causes errors because `public` will always already exist

structure.sql:
```SQL
CREATE SCHEMA auditing;
CREATE SCHEMA other;
CREATE SCHEMA public;
```
error:
```bash
[samps@samps-dus rails-app]$ rails test

psql:/home/samps/sites/DownUnderSports/rails-app/db/structure.sql:16: ERROR:  schema "public" already exists
rails aborted!
failed to execute:
psql -v ON_ERROR_STOP=1 -q -X -f /home/samps/sites/DownUnderSports/rails-app/db/structure.sql down_under_sports_six_test

Please check the output above for any errors and make sure that `psql` is installed in your PATH and has proper permissions.
```

The only way to avoid this problem is to explicitly set *which* schema(s) to dump, or explicitly set `:all`

### System configuration
**Rails version**: 6.0.2.1

**Ruby version**: ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-linux]

### Other Notes
It seems to me, personally, `schema_search_path` causing my other, direct-access-only schemas not to be dumped is not only completely unexpected, but the wrong direction for rails to take; and `:all` should be the default since it's the path of least surprise and also the safest method to not cause people to unknowingly lose data.

If people _want_ to exclude specific schemas, they can set the schemas they want using `config.active_record.dump_schemas` to only include the ones they want or to use `:schema_search_path` if that's how they actually want it to be handled. `schema_search_path` should *only* be used to do what it is actually for by default: setting which objects can be found without a schema qualification; not to exclude schemas from my structure.sql

**Comment by leofin** [comment]
This issue continues to happen.

I'm using Docker and Docker-compose
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]
Rails 6.0.3.4

I generated my app and using the generate scaffold for my project.

I added a new schema to my postgres "billing" set my env:
```ruby
# database.yml
schema_search_path: billing, public
```

```ruby
# application.rb
module Billing
  class Application < Rails::Application
    # Initialize configuration defaults for originally generated Rails version.
    config.load_defaults 6.0

    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration can go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded after loading
    # the framework and any gems in your application.
    config.active_record.schema_format = :sql
  end
end
```

```
# structure.sql
--
-- Name: billing; Type: SCHEMA; Schema: -; Owner: -
--
CREATE SCHEMA billing;
--
-- Name: public; Type: SCHEMA; Schema: -; Owner: -
--
CREATE SCHEMA public;
```

calling `rails test` to run all automatic tests that was generated by scaffold and:
```
psql:/billing/billing/db/structure.sql:23: ERROR:  schema "public" already exists
rails aborted!
failed to execute:
psql -v ON_ERROR_STOP=1 -q -X -f /billing/billing/db/structure.sql billing_test

Please check the output above for any errors and make sure that `psql` is installed in your PATH and has proper permissions.

```

**Comment by vinzenzweber** [comment]
I fixed this issue in my Rails 6.2 app by  adding the `--clean` flag for `pg_dump` and `psql` in `config/application.rb`. The structure.sql file will first delete or drop all tables schemas and so on. In the end the `DROP SCHEMA public;` statement in the resulting structure.sql file fixed the issue.

```
require_relative "boot"

require "rails/all"

# Require the gems listed in Gemfile, including any gems
# you've limited to :test, :development, or :production.
Bundler.require(*Rails.groups)

module MyApp
  class Application < Rails::Application

    # lots of config here

    ActiveRecord::Tasks::DatabaseTasks.structure_dump_flags = ['-v', '--clean']
    ActiveRecord::Tasks::DatabaseTasks.structure_load_flags = ['-v', '--clean']

  end
end

```

**Comment by neontuna** [comment]
Just a note for any other folks like me furiously pasting code they find online but don't actually understand - that `-v` flag is for verbose mode in pg_dump and is going to fill your `structure.sql` file with a bunch of comments that change slightly with every migration.  Therefore irritating your coworkers during code review :)
